src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java

```diff
@@ -968,11 +968,17 @@ void connectionSucceeded(boolean shouldForwardInitialRequest) {
             LOG.debug("Dropping initial request: {}", initialRequest);
         }
 
+        // we're now done with the initialRequest: it's either been forwarded to the upstream server (HTTP requests), or
+        // completely dropped (HTTPS CONNECTs). if the initialRequest is reference counted (typically because the HttpObjectAggregator is in
+        // the pipeline to generate FullHttpRequests), we need to manually release it to avoid a memory leak.
+        if (initialRequest instanceof ReferenceCounted) {
+            ((ReferenceCounted)initialRequest).release();
+        }
     }
```

Comment:
@viacheslav-fomin-main i am not sure we need this
could pls take a look?

Reply:
@ozoz03 I don't know where did this commit come from, looks like you rebased your PR incorrecty.
this was removed in https://github.com/verygoodsecurity/LittleProxy/pull/48, pls revert this change.

Reply:
yeah, I think this is not needed